### PR TITLE
PowerShell Optimizations & Azure CLI MSAL Fix

### PR DESCRIPTION
**Pull request by DCMattyG**
-Updated a few sections of the ArcServersLogonScript.ps1 to be more "PowerShelly"
-Applied a fix to separate the Azure CLI working directories of the ArcServerLogonScript and DataServicesLogonScript to avoid a rare, but possible case of crashing while both attempt to authenticate/refresh and MSAL token simultaneously

### CODE REVIEW

File: azure_jumpstart_arcbox/artifacts/ArcServersLogonScript.ps1 [hunk 1]
```diff
@@ -153,7 +156,7 @@ $authorizedRoles = $roleDefinitions | ForEach-Object { $_ | Where-Object { (Comp
 $hasPermission = $rolePermissions | Where-Object {($_.principalId -eq $spnObjectId) -and ($_.roleDefinitionName -in $authorizedRoles)}
 
 # Copying the Azure Arc Connected Agent to nested VMs
-Write-Output "Copying the Azure Arc onboarding script to the nested VMs"
+Write-Output "Replacing values within Arc Agent install scripts..."
```
Comment (dkirby-ms) [inline azure_jumpstart_arcbox/artifacts/ArcServersLogonScript.ps1:159]:
```suggestion
Write-Output "Replacing values within Azure Arc connected machine agent install scripts..."
```

**Review by dkirby-ms — CHANGES_REQUESTED** [review]
minor naming feedback

**Review by dkirby-ms — APPROVED** [review]
Looks good

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**